### ISSUE TITLE
Which step is jax_compilation_cache_dir expected to speed up?

### ISSUE THREAD

**Issue by jkosinski** [google-deepmind/alphafold3#118]
Which step is expected to be accelerated by jax_compilation_cache_dir? I tried to use it and did not see any difference (tried multiple seeds, multiple runs on the same input json) but maybe I don't know what timing numbers in the log I should look at.

**Comment by jacobjinkelly** [comment]
Setting the jax\_compilation\_cache\_dir avoids re-compiling the model across separate runs of `run_alphafold.py`. See this section in the jax docs for a detailed explanation of [JIT compilation](https://jax.readthedocs.io/en/latest/jit-compilation.html#jit-compiling-a-function).

In AlphaFold 3, JIT compilation is triggered when calling `run_inference` ([permalink](https://github.com/google-deepmind/alphafold3/blob/cdbcf41b71235ec33338edef1f80e03fb0c6bae4/run_alphafold.py#L378)) for the first time on a new [bucket size](https://github.com/google-deepmind/alphafold3/blob/main/docs/performance.md#compilation-buckets). When a JIT compilation is triggered, the timing shown in the logging `"Running model inference for seed ... took ... seconds."` ([permalink](https://github.com/google-deepmind/alphafold3/blob/cdbcf41b71235ec33338edef1f80e03fb0c6bae4/run_alphafold.py#L379-L382)) will be the timing of `JIT compilation + inference`. When there is no new JIT compilation triggered (e.g. when folding a batch of inputs with the same bucket size), the timing shown in the logging will be just the time for model inference.

In our profiling, we found JIT compilation took between 60-85 seconds.

If you would like to debug the compilation cache and make sure it's being written to and read from, we recommend viewing the DEBUG logs for the relevant modules, see the [jax compilation cache docs](https://jax.readthedocs.io/en/latest/persistent_compilation_cache.html#logging-cache-activity).

One way to check the jax compilation cache is enabled is:

1. run with two seeds. The second seed should be faster, as the model is already compiled.  
2. now run again (with any number of seeds). All runs should take the same amount of time as the second seed in the first run, since there will be hits against the jax compilation cache.

**Comment by jkosinski** [comment]
> 1. run with two seeds. The second seed should be faster, as the model is already compiled.
> 2. now run again (with any number of seeds). All runs should take the same amount of time as the second seed in the first run, since there will be hits against the jax compilation cache.

Thanks! I observe 1 but not 2, so it seems I am doing something wrong, but now I know what to check for.
